src/core/system.cpp

```diff
@@ -563,11 +620,16 @@ void run() {
         bool converged = System::step();
         // Maximum error in the fiber derivative
         double fiber_error = 0.0;
-        for (const auto &fib : fc_.fibers) {
-            const auto &mats = fib.matrices_.at(fib.n_nodes_);
-            const Eigen::MatrixXd xs = std::pow(2.0 / fib.length_, 1) * fib.x_ * mats.D_1_0;
-            for (int i = 0; i < fib.n_nodes_; ++i)
-                fiber_error = std::max(fabs(xs.col(i).norm() - 1.0), fiber_error);
+        // XXX FIXME Touch fibers directly, make sure through container rather than here
+        if (fc_->fiber_type_ == FiberContainerBase::FIBERTYPE::FiniteDifference) {
```

Comment:
I know this is baggage from me, but ditto here. `FiberContainer::calc_fiber_error()`